### ISSUE TITLE
`shortest_path` bug in `unitquat_to_rotvec()`

### ISSUE THREAD

**Issue by low5545** [naver/roma#3]
https://github.com/naver/roma/blob/22806dfb43329b9bf1dd2cead7e96720330e3151/roma/utils.py#L255-L280

1. On Line 272, `q1 = -torch.sign(torch.sum(q0*q1, dim=-1, keepdim=True)) * q1` should be `q1 = torch.sign(torch.sum(q0*q1, dim=-1, keepdim=True)) * q1`, since one of the quaternions only need to be flipped when $cos \Omega < 0$, hence dot product $< 0$
2. The `shortest_path` flag is actually useless, since [`unitquat_to_rotvec()`](https://github.com/naver/roma/blob/22806dfb43329b9bf1dd2cead7e96720330e3151/roma/mappings.py#L249) (used on L275) always returns rotation vectors with angles within $[0, \pi]$

**Comment by rbregier** [comment]
Thank you for opening this issue !
Indeed, it seems that the current implementation of this function is bugged:
- As you mentioned, slerp is always performed along the shortest arc here.
- Additionally, `torch.sign` could return 0 in some (rare) cases, leading to a wrong interpolation (i.e. ignoring `q1` and returning `q0`).

These bugs must have appeared during the refactoring of the code, before open-sourcing the library. I will try to fix them in the next release.
Again, thanks for reporting the issue.

**Comment by rbregier** [comment]
Version 1.3.0 should have solved the issue. Thank you for the feedback, and feel free to reopen the issue if needed.

**Comment by low5545** [comment]
@rbregier Thanks for the fix! I realized there is an in-place operation [here](https://github.com/naver/roma/blob/master/roma/mappings.py#L256) in `unitquat_to_rotvec()`, is it better if we use out-of-place operations instead?

**Comment by rbregier** [comment]
Thanks for reviewing changes.
Actually, there is a copy just before this line, so it should not be a problem:
 https://github.com/naver/roma/blob/fa66d5a9095d330b297905ec67ed273601184792/roma/mappings.py#L251
